semantics/src/namespace/types.rs

```diff
@@ -151,6 +169,43 @@ impl Integer {
                 | Integer::I8
         )
     }
+
+    pub fn fits(&self, num: &str) -> bool {
+        let radix = 10;
+
+        fn handle_parse_error<T>(result: Result<T, ParseIntError>) -> bool {
+            if let Err(error) = result {
+                return match error.kind() {
+                    IntErrorKind::PosOverflow => false,
+                    IntErrorKind::NegOverflow => false,
+                    // If we try to parse a negative value for an unsigned type
+                    IntErrorKind::InvalidDigit => false,
+                    // We don't expect this but it would be tragic if we would map this to `false`
+                    // incase it happens because it would mean we sweep a bug under the rug.
+                    other => panic!("Unexpected ParseIntError: {:?}", other),
```

Comment:
@g-r-a-n-t  This is a case where I think we should panic because we never expect it to be hit but if we just map this to `false` and for some reason that branch is hit we would have a hard time searching for a very obscure bug.